pkg/multiTenancyPlugins/naming/nameScoping.go

```diff
@@ -129,3 +131,150 @@ func (nameScoping *DefaultNameScopingImpl) Handle(command utils.CommandEnum, clu
 	}
 	return nil
 }
+
+func CheckContainerReferences(cluster cluster.Cluster, tenantId string, containerConfig *dockerclient.ContainerConfig) error {
+	log.Debugf("CheckContainerReferences containerConfig: %+v", containerConfig)
+	// create arrays of container references to pass to getIDsFromContainerReferences
+	// create volumesFrom array
+	volumesFrom := containerConfig.HostConfig.VolumesFrom
+
+	// create affinity array
+	affinityRef := make([]string, 0)
+	env := containerConfig.Env
+	var affinityIndex int
+	affinityLabelFirst := false
+	for i, e := range env {
```

Comment:
Please don't. Let's talk first - affinity with compose and swarm is very tricky and you are abusing it for different purpose.  The docs are very clear here - if you are using compose version 1 just use links. If you are using version 2 - we would need to put proper overlay network support. 
Note that ordering in Swarm (when deploying a compose template) is not guaranteed.. so yoiu can end up  with failed scheduling. Can you post here how exactly you try to schedule multiple different services on the same node using labels ?